airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -135,22 +139,40 @@ def create_schema(self, schema_json: dict[str, Any]) -> None:
         client = self.conn
         client.schema.create(schema_json)
 
+    @staticmethod
+    def check_http_error_should_retry(exc: BaseException):
+        return isinstance(exc, requests.HTTPError) and not exc.response.ok
+
     def batch_data(
-        self, class_name: str, data: list[dict[str, Any]], batch_config_params: dict[str, Any] | None = None
+        self,
+        class_name: str,
+        data: list[dict[str, Any]] | pd.DataFrame,
+        batch_config_params: dict[str, Any] | None = None,
+        vector_col: str = "Vector",
+        no_retry_attempts_per_object: int = 5,
```

Comment:
```suggestion
        retry_attempts_per_object: int = 5,
```
I am guessing you meant no=number, but looking at the first time I read it as NO (some negation).

Reply:
yes, no=number. Added your suggestion.